File: src/main/java/com/xudong/vam/model/SelectDetail.java
package com.xudong.vam.model;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.NonNull;
import lombok.RequiredArgsConstructor;

@Entity
@Data
@RequiredArgsConstructor
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "t_select_detail")
public class SelectDetail {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NonNull
    @Column(name = "select_id")
    private Long selectId;

    @NonNull
    @Column(name = "root_id")
    private Long rootId;

    @NonNull
    @Column(name = "child_id")
    private Long childId;
}

File: src/main/java/com/xudong/vam/core/model/SelectDetail.kt
package com.xudong.vam.core.model

import jakarta.persistence.*

@Entity
@Table(name = "t_select_detail")
data class SelectDetail(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null,

    @Column(name = "select_id")
    var selectId: Long = 0,

    @Column(name = "root_id")
    var rootId: Long = 0,

    @Column(name = "child_id")
    var childId: Long = 0,
)